src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs

```diff
@@ -48,16 +48,36 @@ internal class ConditionalTestDiscoverer
                     continue;
                 }
 
-                string[] symbols = conditionMemberName.Split('.');
                 Type declaringType = testMethodDeclaringType;
 
-                if (symbols.Length == 2)
+                // We have qualified type name with the assembly name, something like
+                // [ConditionalFact("System.PlatformDetection, CoreFx.Private.TestUtilities!" + nameof(PlatformDetection.IsNonZeroLowerBoundArraySupported))]
```

Comment:
Just to clarify, the qualified type name is "System.PlatformDetection, CoreFx.Private.TestUtilitie" which we can just pass it to Type.GetType without any manipulation. so going with "AssemblyName!ClassName.PropertyName" I'll have to reconstruct the qualified type name from it before I can use it. so I am not seeing any benefit doing that.